Wikipedia/Code/TalkPageContainerViewController.swift

```diff
@@ -38,9 +90,51 @@ class TalkPageContainerViewController: ViewController, HintPresenting {
         return progressController
     }()
     
-    private var repliesAreDisabled = true {
+    private var viewState: TalkPageContainerViewState = .initial {
         didSet {
-            replyListViewController?.repliesAreDisabled = repliesAreDisabled
+            switch viewState {
+            case .initial:
+                self.scrollView?.isUserInteractionEnabled = true
+                navigationItem.rightBarButtonItem?.isEnabled = false
+            case .fetchLoading:
+                fakeProgressController.start()
+                navigationItem.rightBarButtonItem?.isEnabled = false
+            case .fetchInitialResultData:
+                navigationItem.rightBarButtonItem?.isEnabled = false
+                hideEmptyView()
+            case .fetchInitialResultEmpty:
+                navigationItem.rightBarButtonItem?.isEnabled = false
+                wmf_showEmptyView(of: .emptyTalkPage, theme: self.theme, frame: self.view.bounds)
+            case .fetchFinishedResultData:
+                fakeProgressController.stop()
```

Comment:
there's also a `finish()` method on the `fakeProgressController` that sets the progress to `1` before hiding the bar, might be useful for the happy states https://github.com/wikimedia/wikipedia-ios/blob/a1782a596d0cc21925ed24222e59bb53e6967517/WMF%20Framework/FakeProgressController.swift#L71